search_service/proxy/elasticsearch.py

```diff
@@ -377,6 +376,130 @@ def fetch_table_search_results(self, *,
+                            'OR name:({name}) ' \
+                            'OR schema_name:(*{name}*) ' \
+                            'OR schema_name:({name}) ' \
+                            'OR description:(*{name}*) ' \
+                            'OR description:({name}) ' \
+                            'OR column_names:(*{name}*) ' \
+                            'OR column_names:({name}) ' \
+                            'OR column_descriptions:(*{name}*) ' \
+                            'OR column_descriptions:({name}))'.format(name=query_term)
+
+        if not query_dsl and not add_query:
+            raise Exception('Unable to convert parameters to valid query dsl')
+
+        result = ''
+        if query_dsl and add_query:
+            result = query_dsl + ' AND ' + add_query
+        elif add_query and not query_dsl:
+            result = add_query
+        elif query_dsl and not add_query:
+            result = query_dsl
+
+        return result
+
+    @timer_with_counter
+    def fetch_table_search_results_with_filter(self, *,
+                                               query_term: str,
+                                               search_request: dict,
+                                               page_index: int = 0,
+                                               index: str = '') -> SearchResult:
+        """
+        Query Elasticsearch and return results as list of Table objects
+        :param search_request: A json representation of search request
+        :param page_index: index of search page user is currently on
+        :param index: current index for search. Provide different index for different resource.
+        :return: SearchResult Object
+        """
+        current_index = index if index else \
+            current_app.config.get(config.ELASTICSEARCH_INDEX_KEY, DEFAULT_ES_INDEX)
+        if not search_request:
+            # return empty result for blank query term
+            return SearchResult(total_results=0, results=[])
+
+        try:
+            query_string = self.convert_query_json_to_query_dsl(search_request, query_term)
+        except Exception as e:
+            LOGGING.exception(e)
+            # return nothing if any exception is thrown under the hood
+            return SearchResult(total_results=0, results=[])
+        s = Search(using=self.elasticsearch, index=current_index)
+
+        query_name = {
+            "function_score": {
+                "query": {
+                    "query_string": {
+                        "query": query_string
+                    }
+                },
+                "field_value_factor": {
+                    "field": "total_usage",
+                    "modifier": "log2p"
+                }
+            }
+        }
+
+        return self._search_helper(page_index=page_index,
+                                   client=s,
+                                   query_name=query_name,
+                                   model=Table)
```

Comment:
nit: the model here should match the index being used, right?

Reply:
agree

Reply:
Are changes needed here?

Reply:
instead of hard to table, use the index from input argument. Maybe worth putting a note/todo later to refactor this function to work all datasets cases and not just table.